lib/rubygems/specification.rb

```diff
@@ -103,6 +103,13 @@ class Gem::Specification < Gem::BasicSpecification
      4 => 18,
   }.freeze
 
+  ##
+  # Some Specification methods/properties are deprecated and then removed.
+  # Gems installed with older RubyGems versions may have these items
+  # in their .gemspec files.  Used in #method_missing to ignore them.
+
+  REMOVED_METHODS = [:rubyforge_project=].freeze
```

Comment:
I'm not sure if we want to maintain a list of these.

Maybe we should just remove it in the next release since we already warned that it will be deprecated at `2019, 12`  🤔

Reply:
But then things will crash in the case @MSP-Greg mentioned, right? I don't think we want to exchange annoying warnings with more annoying crashes, that can't really be fixed easily.

I believe the approach in #3047 is better though, because if the author of the offending gemspec ever rebuilds the gem, she will notice the deprecation needs to be addressed?

Reply:
Was awake anyway... 

> But then things will crash

Even to the point where one can't uninstall the offending gem.

Re https://github.com/rubygems/rubygems/pull/3047, I left that open because I personally wondered about a combination of the two. and also https://github.com/rubygems/rubygems/pull/3047 doesn't allow for actually removing the code for the setter method/attribute.

One question re https://github.com/rubygems/rubygems/pull/3047, it based the 'build' logic on actually using BuildCommand.  Is there a better way to detect that?

> the author of the offending gemspec ever rebuilds the gem, they will notice the deprecation needs to be addressed?

I also wonder about whether it should be a 'stop' rather than a notice/warning.

Thanks, & AFK...

Reply:
Yeah. At some point we should raise instead of warn. And at some point after changing the code to raising, we'll be able to completely remove the code.

But for now we just want to fix the issue where deprecation are being printed where not intented, and I think the right way to do that is #3047 better than this PR.

Reply:
What I was trying to say above is (once a determination is made about how to determine if a build is occurring) that the code here could incorporate that into the `#method_missing` logic, and allow removing code.

Something like changing:
```ruby
return if REMOVED_METHODS.include? sym
```
to
```ruby
return if REMOVED_METHODS.include?(sym) and
  Gem::CommandManager.instance.current_command != :build
```

And then it will raise only on build.  Obviously, #3047 needs to be finished first...